components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -86,7 +87,11 @@ public String addTenant(Tenant tenant) throws TenantMgtException {
 
             // For the super tenant tenant creation, tenants are always activated as they are created.
             TenantMgtUtil.activateTenantInitially(tenantInfoBean, tenantId);
+
             if (INVITE_VIA_EMAIL.equalsIgnoreCase(tenant.getProvisioningMethod())) {
+                if (tenant.getClaimsMap() == null) {
```

Comment:
Shall we fix this at tenant object level? It should not return null in the tenant.getClaimsMap() API

Reply:
ok fixed with https://github.com/wso2/carbon-kernel/pull/2683

Reply:
Fixed with https://github.com/wso2/carbon-kernel/pull/2683